corehq/pillows/app_submission_tracker.py

```diff
@@ -36,21 +37,28 @@ def get_app_form_submission_tracker_pillow(pillow_id='AppFormSubmissionTrackerPi
 
 
 class AppFormSubmissionReindexDocProcessor(BaseDocProcessor):
-    def __init__(self, pillow_processor):
+    def __init__(self, pillow_processor, data_source_type, data_source_name):
         self.pillow_processor = pillow_processor
+        self.data_source_type = data_source_type
+        self.data_source_name = data_source_name
 
     def process_doc(self, doc):
-        change = self._doc_to_change(doc)
-        self.pillow_processor.process_change(None, change)
+        change = self._doc_to_change(doc, self.data_source_type, self.data_source_name)
+        try:
+            self.pillow_processor.process_change(None, change)
```

Comment:
this feels weird to me. i almost feel like the change should be here: https://github.com/dimagi/commcare-hq/blob/br/fix-app-form-submissions-pillow/corehq/util/doc_processor/interface.py#L208

and instead of relying on a boolean to be returned from a `process_change` function, have it be `ok` if no exceptions are thrown. not sure if we're too dependent on this behavior already or what the original intent was